File: backend/src/main/java/br/com/ifsp/classify/dtos/create/EmployeeCreateDTO.java
package br.com.ifsp.classify.dtos.create;

import java.time.LocalDate;
import java.util.List;

public record EmployeeCreateDTO(
        String name,
        LocalDate birthDate,
        String cpf,
        LocalDate hireDate,
        UserCreateDTO user,
        List<TelephoneCreateDTO> telephones
) {}

File: frontend/src/shared/dtos/employees/EmployeeCreateDTO.ts
import type { UserCreateDTO } from "../user/UserCreateDTO";
import type { TelephoneCreateDTO } from "../telephone/TelephoneCreateDTO";

export interface EmployeeCreateDTO {
    name: string;
    birthDate: string;
    cpf: string;
    hireDate: string;
    user: UserCreateDTO;
    telephones: TelephoneCreateDTO[];
}
